internal/operator/snapshotlease/snapshotlease.go

```diff
@@ -0,0 +1,200 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package snapshotlease
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetSnapshotLease indicates an error in getting the snapshot lease resources.
+	ErrGetSnapshotLease druidv1alpha1.ErrorCode = "ERR_GET_SNAPSHOT_LEASE"
+	// ErrSyncSnapshotLease indicates an error in syncing the snapshot lease resources.
+	ErrSyncSnapshotLease druidv1alpha1.ErrorCode = "ERR_SYNC_SNAPSHOT_LEASE"
+	// ErrDeleteSnapshotLease indicates an error in deleting the snapshot lease resources.
+	ErrDeleteSnapshotLease druidv1alpha1.ErrorCode = "ERR_DELETE_SNAPSHOT_LEASE"
+)
+
+type _resource struct {
+	client client.Client
+}
+
+// New returns a new snapshot lease operator.
+func New(client client.Client) component.Operator {
+	return &_resource{
+		client: client,
+	}
+}
+
+// GetExistingResourceNames returns the names of the existing snapshot leases for the given Etcd.
+func (r _resource) GetExistingResourceNames(ctx component.OperatorContext, etcd *druidv1alpha1.Etcd) ([]string, error) {
```

Comment:
This should not be explicitly mentioned in the docstring, since the names of the leases depend entirely on the `Etcd` resource name. This method simply returns the names of the existing snapshot leases for this component, whichever they are.